yktrace/src/tir.rs

```diff
@@ -49,44 +49,13 @@ impl<'a, 'm> TirTrace<'a, 'm> {
         // introducing it via `trace_locals()`.
         let mut defined_locals: HashSet<Local> = HashSet::new();
         let mut def_sites: HashMap<Local, usize> = HashMap::new();
-        let mut last_use_sites = HashMap::new();
+        let mut live_vars: HashSet<Local> = HashSet::new();
```

Comment:
Potential optimisation for later? Use a bitset?

Reply:
What's the type of `Local`? I notice `defined_locals` is also a `HashSet<Local>`, for example.

Reply:
I can do it here if you like.

Reply:
`Local` is a wrapper around `u32`.

Reply:
In `yk` we might as well use https://crates.io/crates/vob which is faster for some use case than bitset -- assuming that Vob does work well here! [Using Vob inside ykrustc is something we decided against, but in an arbitrary library, we might as well use the most appropriate library.]

Reply:
So make a set and then convert it to a `vob`? We don't want any duplicates.

Reply:
A `Vob` is effectively a set, if you initialise it with a size (this does assume you know the maximum size; if you don't, then you either need a wrapper around vob, or to use bitset).

Reply:
Up to you.

Reply:
I think you should try bitset and vob and see which works best in your opinion.

Reply:
Ah yes. I was thrown by "vector" for a moment. It's just that the set is represented by a vector (of bits).

Reply:
I'm not sure if we know the max size ahead of time. I leave that to Lukas to look at :)

Reply:
I think we should know the size of all locals (though not the live ones) from the SIR. I'll have a look. The live locals are gonna be a subset of all locals though, so not sure if that would waste space when using a vob.

Reply:
If, as is likely, you have 64 or fewer locals, you won't waste any memory at all (since each local will be represented by a single bit).

Reply:
Hmm, doesn't look like we currently have a way to get the total number of used locals (though we could add the in the future). I also looked into `BitSet` which is sort-of unmaintained at the moment: https://crates.io/crates/bit-set. Thoughts?

Reply:
Since this is an optimisation, I'd say don't worry about it now, or do it in another PR. I have a PR blocked on this one, so let's get it in.

Reply:
Agreed!